src/pages/App.tsx

```diff
@@ -146,6 +149,10 @@ export default function App() {
     setScrolledState(false)
   }, [pathname])
 
+  useEffect(() => {
+    setShouldDisableNFTRoutes(Boolean(search && search.includes('disableNfts=true')))
+  }, [search, setShouldDisableNFTRoutes])
```

Comment:
Why do we need an atom here? Can't we simply use `useLocation` in a NavBar component and move this logic there?

Reply:
the effect could be run anywhere, since it's just consuming react-router-dom

The atom is used because the `shouldDisableNFTRoutes` state is consumed at the NavBar to disable user interaction and also at the nft-related `Route` components below to disable further direct nav.